__tests__/proxy-sse-test.ts

```diff
@@ -0,0 +1,204 @@
+  onmessage?: (evt: Event, resolve: (...any) => void) => void;
+  onerror?: (evt: Event, resolve: (...any) => void) => void;
+  eventListeners?: IEventListener[];
+}
+
+function sse(
+  port: number,
+  callbacks: ICallbacks = {},
+  options: object = {}
+): Promise<any> {
+  return new Promise(resolve => {
+    const source = new EventSource(`http://localhost:${port}`, options);
+
+    if (callbacks.onmessage) {
+      source.onmessage = evt => {
+        callbacks.onmessage(evt, resolve);
+      };
+    } else if (callbacks.onerror) {
+      source.onerror = evt => {
+        callbacks.onerror(evt, resolve);
+      };
+    } else {
+      source.onopen = () => resolve(source);
+    }
+
+    (callbacks.eventListeners || []).forEach(({ type, cb }) =>
+      source.addEventListener(type, cb)
+    );
+  });
+}
+
+describe("Proxy - Server Sent Events", () => {
+  let s;
+  let msController;
+  let msPort;
+
+  beforeEach(async () => {
+    s = await createServer();
+    msController = new MockServerController(s.port);
+    msPort = await msController.start();
+
+    s.app.use((req, res, next) => {
+      res.writeHead(200, {
+        "Content-Type": "text/event-stream",
+        "Cache-Control": "no-cache",
+        Connection: "keep-alive"
+      });
+
+      res.sseSendUntyped = data => {
+        res.write("data: " + JSON.stringify(data) + "\n\n");
+      };
+
+      res.sseSendTyped = (type, data) => {
+        res.write("event: " + type + "\n");
+        res.write("data: " + JSON.stringify(data) + "\n\n");
+      };
+
+      next();
+    });
+  });
+
+  afterEach(() => {
+    msController.stop();
+    s.server.close();
+  });
+
+  describe("Requests", () => {
+    let mockMiddleware;
+    beforeEach(() => {
+      mockMiddleware = jest.fn();
+      s.app.use((req, res) => {
+        mockMiddleware(req);
+        res.sseSendUntyped({ done: true });
+        res.end();
+      });
+    });
+
+    it("opens a connection", async () => {
+      const source = await sse(msPort);
+
+      source.close();
+      expect(mockMiddleware).toHaveBeenCalled();
+    });
+
+    it("sends the headers", async () => {
+      const source = await sse(msPort, {}, { headers: { "my-header": "foo" } });
+
+      source.close();
+      const call = mockMiddleware.mock.calls[0][0];
+      expect(call.headers["my-header"]).toBe("foo");
+    });
+  });
+
+  describe("Response", () => {
+    it("forwards data", async () => {
+      s.app.use((req, res) => {
+        res.sseSendUntyped({ count: 1 });
+        res.sseSendUntyped({ count: 2 });
+        res.end();
+      });
+
+      const mockOnMessage = jest.fn();
+      let callCount = 0;
+      await sse(msPort, {
+        onmessage: (evt, resolve) => {
+          callCount++;
+          mockOnMessage();
+          if (callCount === 2) {
+            resolve();
+          }
+        }
+      });
+      expect(mockOnMessage).toHaveBeenCalledTimes(2);
+    });
+
+    it("forwards data with a similar interval", async () => {
```

Comment:
🗣  …in same order…?

Reply:
otherwise I'd check for a range between 700-900, not >200? 🤔

Reply:
I don't check for the order, need to do another test then